### DISCUSSION TITLE
Fetching a draft post by slug

### DISCUSSION THREAD

**Discussion by sebpowell** [sanity-io/sanity#4535] [Sanity Studio Support] [answered]
I'm in the process of implementing preview mode with Next.js 13, and I'm not sure how to go about fetching a post that it's in draft mode. It all works fine when the post is published, but that seems to defeat the purpose of preview mode. 

This is the query I'm using:

```groq
*[_type == "post" && slug.current == $slug][0]{ ..., authors[]->, categories[]->]
```

How would I adapt this to fetch the post when it's in unpublished? 

Many thanks in advance.

**Comment by Codeshark-NET** [comment, marked as answer]
Hey there @sebpowell! 👋

You can include/exclude drafts by filtering with the drafts path. 

**Here's all posts without drafts:**
`*[_type == "post" && !(_id in path('drafts.**'))]`

**Here's only the drafts:**
`*[_type == "post" && (_id in path('drafts.**'))]`

➡️ Have a look here for more information: https://www.sanity.io/docs/drafts#ebfb408fa9b6

Cheers!

**Reply by sebpowell** [reply]
Great, thank you @Codeshark-NET!

**Reply by Rhym** [reply]
Just an update to this, you now need to swap out `_id` for `_originalId`.
